aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1079,6 +1079,12 @@ Tensor trace_cpu(const Tensor& self) {
   return result;
 }
 
+// TODO: this routine should be implemented without diag and sum for perf problems,
+// see https://github.com/pytorch/pytorch/pull/47305,
+Tensor linalg_trace(const Tensor& self, int64_t offset) {
```

Comment:
This is also missing a `TORCH_CHECK(self.dim() >= 2, ...`

Reply:
I think it'd also be nice to check that offset is in the range [-dim, dim) and throw a nice error message otherwise.